exercises/exercise_2/README.md

```diff
@@ -1,219 +1,195 @@
+
+    ```bash
+    $ cat voting-app-merged.dockerapp
+    <metadata.yml contents>
+    ---
+    <docker-compose.yml contents>
+    ---
+    <parameters.yml contents>
+    ```
+
+## Splitting the Docker Application
+
+Once merged, the application can be split back into separate files using the `docker app split` command.
+
+1. Run the `docker app split` command to split the file back out into separate files
+
+    ```bash
+    $ docker app split voting-app-merged.dockerapp -o voting-app-split.dockerapp
+    ```
+
+2. Look in the `voting-app-split.dockerapp` directory and you'll see the files split apart.
+
+    ```bash
+    $ ls voting-app-merged.dockerapp
+    docker-compose.yml  metadata.yml        parameters.yml
+    ```
+
+3. At this point, we don't need the split and merged versions anymore. Let's remove them to reduce confusion going forward.
+
+    ```bash
+    $ rm -r voting-app-merged.dockerapp voting-app-split.dockerapp
+    ```
+
+
+## Inspecting our Docker Application
+
+We can use the `docker app inspect` command to get a quick output of all of the services, number of replicas, ports, and the image being used.
+
+1. Run the `docker app inspect` command to inspect our application. You shouldn't need to provide the app name, as there is only one in the directory.
+
+    ```bash
+    $ docker app inspect
+    voting-app 0.1.0
+
+    Maintained by: dapworkshop <dapworkshop@docker.com>
+
+    Voting App
+
+    Services (6) Replicas Ports   Image
+    ------------ -------- -----   -----
+    vote         2        5000    mikesir87/examplevotingapp_vote
+    redis        1                redis:alpine
+    worker       1                dockersamples/examplevotingapp_worker
+    db           1                postgres:9.4
+    results      1        5001    mikesir87/examplevotingapp_result
+
+    Networks (3)
+    ------------
+    frontend
+    backend
+    proxy
+
+    Volume (1)
+    ----------
+    db-data
+    ```
+
+
+
+## Validating our Application
+
+Before we're ready to ship our application, we should validate it to make sure everything is set. Specifically, validation does the following:
+
+- Ensures our compose file is valid (correct syntax, etc.)
+- Ensures required metadata is provided (name, version) and is the correct format
+- Ensures all parameters (which we'll talk about next) have default values
+
+1. Run the `docker app validate` command to make sure our application is valid.
+
+    ```bash
+    docker app validate
+    Validated "/root/voting-app.dockerapp"
+    ```
+
+2. Let's make a change to invalidate the application. In the `voting-app.dockerapp/metadata.yml` file, comment out the `version` field using a `#`. Then, revalidate.
+
+    ```bash
+    docker app validate
+    failed to validate metadata:
+    - version: version is required
+    ```
+
+    We have an error! Go ahead and fix it and revalidate.
+
+
+## Recap
```

Comment:
I think this Recap part is missing on exercise 1? 🤔

Reply:
I'll go ahead and remove it, as it's mostly a copy of the objectives.